cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java

```diff
@@ -435,6 +436,10 @@ private void httpExceptionHandler(Promise<? extends CruiseControlResponse> resul
             // Vert.x throws a NoStackTraceTimeoutException (inherits from TimeoutException) when the request times out
             // so we catch and raise a TimeoutException instead
             result.fail(new TimeoutException(t.getMessage()));
+        } else if (t instanceof NoRouteToHostException) {
+            // Netty throws a AnnotatedNoRouteToHostException (inherits from NoRouteToHostException) when it cannot resolve the host
+            //so we catch and raise a NoRouteToHostException instead
```

Comment:
I still don't get from this why are you not checking for `t instanceof AnnotatedNoRouteToHostException` => that is what the comment needs to explain. Because you say you do something else than what you actually do.

Reply:
We are catching `NoRouteToHostException` just like we catch the ConnectException and not the derived AnnotatedConnectException. I guess in the comments I wrote the same reason that it catch and raise a `NoRouteToHostException`  instead.

Reply:
But the comment does not really explain it. This seems weird:
* Why do you need to raise another exception when it inherits?
* If you need to raise the `NoRouteToHostException` instead, why do you check for `instanceof NoRouteToHostException` and not `instanceof AnnotatedNoRouteToHostException`

None of that is explained there. The comment just says what it does, but it needs to say why it does it.

Reply:
Actually if you see the other if conditions there also we are doing the same thing. We try to catch the `Parent Exception`. We are doing this because `ConnectException`, `NoRouteToHostException` all these exceptions are basically part of of Java base.

Reply:
I'm not sure that is a justification - that some other place does the same for sure not. And if an exception from some undesired transitive dependency is raised, it is probably a bug somewhere. But in any case, it is not what the comment says. So even if the code is kept as it is, the comment should be updated.

Reply:
Yeah I agree with you. I guess its better to catch `AnnotatedNoRouteHostException` itself and then raise a `NoRouteToHostException`. I was just trying to be consistent. I will fix the comment here

Reply:
I think you should find out who wrote the original code and have them explain why they did what they did.

Reply:
@tomncooper do you remember why we went this way? Was it just for handling the Java base exceptions up the chain instead of handling the Netty related ones?

Reply:
I do not remember why this was done? Git blame shows I wrote and Paolo was the last to change it. It doesn't seem to make sense to rethrow the parent Exception?

Reply:
Maybe it was done for making the code which catches that exception more general and not relying on some specific exception coming from Netty but the base one coming from Java?

Reply:
Well if you want to handle this you could still catch the more generic exception? I tried deleting this code and running the tests and nothing bad occured (except one test that exects the connection exceception not the netty one to be thrown). Maybe the system tests would show more, but this does seem to be a pointless operation?

Reply:
Should I catch the exception which are thrown itself and not the parent Exceptions then @ppatierno @tomncooper . Also I will test if it works fine and if it does then maybe we can keep it that way?

Reply:
+1 for not doing this re-throwing of the parent. I tested the change and nothing broke so I am for removing this.

The only reason for doing this I can see is not exposing the dependency exception types in the API but they inherit from base Java exceptions so this is nor eeally an issue as far as I am concerned.